examples/petsc/Makefile

```diff
@@ -22,8 +22,8 @@ ifneq ($(CUDA_LIB_DIR),)
   LDLIBS += -lcudart -lnvrtc -lcuda
```

Comment:
Are you sure this is necessary? I wouldn't expect the example to call CUDA directly (it would be via libceed.so, which is already linked to the CUDA libs). I'd rather keep this complexity out of this makefile if possible.

Reply:
@valeriabarra

Reply:
@jedbrown on CUSummit I wasn't even able to compile libCEED to find lcuda without adding that stubs. I tested it by creating a folder from scratch, pulling libCEED from master, and load the default intel compiler and cuda modules and simply running `make`. I get the following:
```          
CC build/backends/cuda-shared/ceed-cuda-shared.o
icc: command line warning #10006: ignoring unknown option '-ffp-contract=fast'
icc: command line warning #10006: ignoring unknown option '-fopenmp-simd'
        NVCC build/backends/cuda/cuda-vec.o
        NVCC build/backends/cuda-reg/cuda-reg-basis.o
          CC lib/libceed.so
ld: cannot find -lcuda
```
I think on CU Summit they don't have the CUDA drivers directly installed on the compile node

Reply:
Tested this on CU Summit. With the top level Makefile addition of CUDA_LIB_DIR_STUBS, libCEED compiles and links to cuda fine. In fact, when querying `readelf -d lib/libceed.so` it gives me:
```
Library runpath: [/curc/sw/cuda/9.1.85/lib64:/curc/sw/cuda/9.1.85/lib64/stubs:/curc/sw/intel/17.4/impi/2017.3.196/intel64/lib/release_mt:/curc/sw/intel/17.4/impi/2017.3.196/intel64/lib:/opt/intel/mpi-rt/2017.0.0/intel64/lib/release_mt:/opt/intel/mpi-rt/2017.0.0/intel64/lib]
```
But when trying to build the petsc examples with `make build/petsc-bps` commenting out lines 14-23 above, I get the following error:
```
ld: warning: libcuda.so.1, needed by /projects/vaba3353/CUDA_CEED/libCEED/lib/libceed.so, not found (try using -rpath or -rpath-link)
/projects/vaba3353/CUDA_CEED/libCEED/lib/libceed.so: undefined reference to `cuLaunchKernel'
/projects/vaba3353/CUDA_CEED/libCEED/lib/libceed.so: undefined reference to `cuModuleUnload'
/projects/vaba3353/CUDA_CEED/libCEED/lib/libceed.so: undefined reference to `cuModuleLoadData'
/projects/vaba3353/CUDA_CEED/libCEED/lib/libceed.so: undefined reference to `cuGetErrorName'
/projects/vaba3353/CUDA_CEED/libCEED/lib/libceed.so: undefined reference to `cuModuleGetFunction'
make[1]: *** [bps] Error 1
```
So that's why I had to add those lines.

Reply:
But all the other tests and examples worked without this? That sounds fishy. I'll try to reproduce.

Reply:
I tried also `make build/mfem-bp1` and `make build/mfem-bp3`, and same error message

Reply:
This seems like what @YohannDudouit was trying to work around by adding the "stubs" directory when building libceed.so. I've reproduced the issue on Summit, but I don't have a simple solution.

The CUDA runtime API does not link directly to the driver API (I assume it uses `dlopen`) to handle exactly this issue. If we're using the driver API, we may need to do that. It's desirable for someone to be able to build libceed with GPU support and run non-GPU backends on nodes that do not have GPUs. The driver generally won't be installed on such nodes.

But that's an issue for another PR. I'd prefer to remove this hack to get PETSc compiles going because we'd need to do it everywhere and I think that's unacceptable as a long-term linking interface. So temporarily until we resolve this issue, let's say libCEED doesn't support building on nodes without the drivers. Is that okay with you?

Reply:
I think you mean this [commit](https://github.com/CEED/libCEED/pull/249/commits/5c8aa7592d9944c593dbf91b123a6bbc5dfbdfa5) for the "stubs" directory? 

Anyhow, I can see that it is less desirable to mantain this way---that was something that got me going for the testing on CU Summit, but we can remove it and take care of it in a separate PR.